rasa/engine/recipes/graph_recipe.py

```diff
@@ -36,17 +36,23 @@ def graph_config_for_recipe(
                 "Add configuration to the recipe itself if you want them to be used.",
                 docs=DOCS_URL_GRAPH_RECIPE,
             )
-        # TODO: Add core_target and nlu_target to graph configuration options so we
-        # don't force graph node names to be `run_RegexMessageHandler` and
-        # `select_prediction`. These are used in validation and during execution.
-        core_target = None if training_type == TrainingType.NLU else "select_prediction"
+        # Note that default recipe has `nlu_target` and `core_target` as
+        # fixed values of `run_RegexMessageHandler` and `select_prediction`
+        # respectively. For graph recipe, target values are customizable. These
+        # can be used in validation (default recipe does this validation check)
+        # and during execution (all recipes use targets during execution).
+        if training_type == TrainingType.NLU:
+            core_target = None
+        else:
+            core_target = config.get("core_target", "select_prediction")
```

Comment:
Maybe it's better to fail fast rather than have a default?

Reply:
I'm more inclined towards "fail-safe" rather than "fail-fast" but @ka-bu had a similar opinion on another line where if config file changes between Rasa reading it and auto-configuring that we should fail instead of warning the user. She has a comment on this pull request. I feel defensive programming is better here as this is also the default target for default recipe? We could also add a warning log as a sort of a trade off? Otherwise I can also fail it :) Not too strong on this.

Reply:
I'm just thinking from a user perspective:
1. User decides to create a new graph recipe config.
2. They forget to add the targets
3. They get an error saying that component `select_prediction` or `RegexMessageHandler` does not exist in the schema.
4. They assume they are missing a required component, when actually the just need to configure the targets to fit their components.

Reply:
Added code to raise with a message if targets are missing.